### PR TITLE
doc: nginx websocket config (#2140)

### PR DESCRIPTION
**Pull request by karlsson**
### Description

Fix #[2140]

Add documentation for proxying MQTT transport on websocket with nginx.

### Checklist

- [x] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

File: doc/developer-guide/deployment/nginx.rst [hunk 1]
```diff
@@ -12,6 +12,10 @@ are planning to server (the ``user/sites/your_site/config``
 files). The ``hostname`` value should not contain any port number, if
 you run from port 80: ``{hostname, "test.zotonic.com"}``.
 
+Zotonic also makes use of a websocket connection for MQTT messages
+under the ``/mqtt-transport`` location and so you have to add an extra
+proxy directive for this.
+
 Below is a configuration file we use to proxy nginx to zotonic. Be
 sure to replace all occurrences of ``test.zotonic.com`` with your own
 hostname::
```
Comment (mworrell) [inline doc/developer-guide/deployment/nginx.rst]: I am not sure why the doc tool suddenly complains, but apparently the default code-syntax is 'erlang'.

We need to change this with:

```
.. code-block:: nginx
```

(And replace the double ':' after "hostname::" with a single ':')

Comment (karlsson) [reply doc/developer-guide/deployment/nginx.rst]: Thanks Marc,
After checking the documentation on how to generate documentation (apt install sphinx-common) I am able to check before commit now. :smile: 
Will fix.

Comment (mworrell) [reply doc/developer-guide/deployment/nginx.rst]: Then you can do better than me, somehow my local docs build is a bit broken.

Comment (karlsson) [reply doc/developer-guide/deployment/nginx.rst]: Well I think I was a bit confused with different python versions and pip install as per the docs. So I ended up just installing sphinx-common (sudo apt-get install sphinx-common). And I think I am on python2.7 now. It complains a bit but generation seems to work OK, I think.
/home/mikael/.local/lib/python2.7/site-packages/sphinx/builders/html.py:1171: RemovedInSphinx20Warning: Now html_sidebars only allows list of sidebar templates as a value. Support for a string value will be removed at Sphinx-2.0.
  self.add_sidebars(pagename, ctx)
..
build succeeded.
The HTML pages are in _build/html.

**Comment by mworrell** [comment]
Great, it passes the build.  Thanks!

**Pull request merged**